Apache Cassandra chapter

JohnAZoidberg opened the pull request:
Hi, I'd like to contribute to the book, please :)

lschuermann commented on Makefile:
```diff
@@ -0,0 +1,9 @@
+.PHONY: clean
+
+main.pdf: main.tex content/* *.bib glossary-entries.tex
+	latexmk -pdflatex="pplatex -c pdflatex --" --shell-escape -pdf -interaction=nonstopmode main.tex 2>&1 | tee latexmk_log.txt
```
We already discussed in #2 that adding a Makefile in this form forces people to use applications like `pplatex`. Please remove it.

JohnAZoidberg replied:
Adding a Makefile doesn't force anybody to do anything.

JohnAZoidberg replied:
Pandoc was the problem, that's why we decided to close that PR.

lschuermann replied:
That may be true. After a quick discussion - and for various reasons, we've decided to not introduce a makefile.

JohnAZoidberg replied:
I'd like you to put forward your reasons.
Do those same reasons apply to the `default.nix` file?

JohnAZoidberg replied:
I removed it, I'll not be able to build it to test it anymore, however - the Nix syntax is beyond me.

lschuermann replied:
> I'd like you to put forward your reasons.

We are using `nix` for continuous integration purposes and for having a identical, verified & controlled build environment. We are not going to maintain / support any other way to build the project in the upstream repository. What you do locally is your decision.


> I removed it, I'll not be able to build it to test it anymore, however - the Nix syntax is beyond me.

Then you have a problem, I suppose. Apparently, you somehow managed to come this far and the CI successfully builds the project - if you have specific issues or questions, we're always here to help you. :)

We won't discuss this further.

lschuermann commented on cassandra.bib:
```diff
@@ -0,0 +1,518 @@
+% https://www.cl.cam.ac.uk/~ey204/teaching/ACS/R212_2014_2015/papers/lakshman_ladis_2009.pdf
```
Please don't add a separate bibliography file. This could confuse people and makes de-duplication harder.
We'll have to figure something out eventually if the file grows too large. Adding a separate file for one chapter is not an option.

lschuermann commented on content/3_0_cassandra.tex:
```diff
@@ -0,0 +1,592 @@
+\chapter{Apache Cassandra}
```
The intention was to keep the introduction and specifics about Cassandra separate. However, as we're only analyzing one column-oriented database, I'm fine with having one chapter. Please change the title to reflect this - like the GraphDB / Neo4J group did.

lschuermann commented on content/cassandra.md:
```diff
@@ -0,0 +1,541 @@
+\chapter{Cassandra}
```
I'm fine with the markdown source being included here. However, please change the name in accordance to our standard (outlined in the readme).

lschuermann commented on glossary-entries.tex:
```diff
@@ -54,3 +51,26 @@
 \newacronym{hdd}{HDD}{hard disk drive}
 \newacronym{ecc}{ECC}{error correcting code}
 \newacronym{ascii}{ASCII}{American Standard Code for Information Interchange}
+\newacronym{api}{API}{Application Programming Interface}
```
In a previous PR we've decided to change this to lowercase. Please outline why this change is required - it might look weird in other texts now.

JohnAZoidberg replied:
So what about all other acronyms? Should everything be lowercase?
I for one think it looks terrible.

lschuermann replied:
The correct way to do this is to have the acronym's first usage in the text as lowercase, and in the glossary as uppercase (except for names of course).
Therefore, please change your added entries to be all lowercase. We'll make the first character appear uppercase in another PR.

lschuermann replied:
The correct way to do this would be to have the acronym's first usage in the text as lowercase, and in the glossary as uppercase (except for names of course).
Therefore, please change your added entries to be all lowercase. We'll make the first character appear uppercase in another PR.

JohnAZoidberg replied:
I did it but can you please cite a reference which says that this is the correct way?

lschuermann replied:
This was discussed with @danielrutz - we both agreed on this. Unfortunately, I can not provide you a reference that says so.

lschuermann commented on main.tex:
```diff
@@ -41,6 +42,8 @@
   style=apa
 ]{biblatex}
 \addbibresource{bibliography.bib}
+\addbibresource{cassandra.bib}
```
Please remove.

lschuermann commented on cassandra.bib:
```diff
@@ -0,0 +1,518 @@
+  howpublished = {\url{https://docs.datastax.com/en/dse/6.7/cql/}}
+}
+
+@misc{cassandra3cqldoc,
```
This entry appears twice.

lschuermann commented:
https://travis-ci.org/dhbw-stginf16a/nosql-book/builds/520989650#L4308

You've got some bad boxes with a badness of `10000`. While these don't necessarily have to be breaking anything - and aren't a strict requirement to getting your chapter merged - you could try fixing these.

JohnAZoidberg commented:
Addressed all concerns I unconditionally agree with.

@jpyen seems like your graphics cause a lot of underfull vboxes, I don't want to decide how to fix that.

JohnAZoidberg commented:
There's one hbox that's overfilled by >60pts. One of our URls is way too long and bibtex doesn't break it.
The `url` package seems to address that issue but I'll leave that to the editors:
- https://tex.stackexchange.com/questions/115690/urls-in-bibliography-latex-not-breaking-line-as-expected
- https://tex.stackexchange.com/questions/59122/how-to-avoid-using-sloppy-document-wide-to-fix-overfull-hbox-problems

I'll leave it to the editors to decide how to solve this, might be a problem for other chapters too.

lschuermann commented:
> The `url` package seems to address that issue but I'll leave that to the editors

Yes, the `url` package seems to address this issue. Please go ahead and add it. :+1:

jpyen commented:
> > The `url` package seems to address that issue but I'll leave that to the editors
>
> Yes, the `url` package seems to address this issue. Please go ahead and add it.

Thank you for the opportunity, we choose not to change it.

Kind regards.

JohnAZoidberg commented:
Fixed. We don't even need the URL package (couldn't get it to work anyways).

lschuermann commented on main.tex:
```diff
@@ -72,7 +77,7 @@
 
 \title{NoSQL Databases}
-\author{Daniel Rutz and Leon Schürmann (Eds.)}
+\author{Class StgInf16A of DHBW Stuttgart}
```
If you insist on changing this, please spell the course abbreviation correctly:
```suggestion
\author{Course STG-TINF16A of the Baden-Wuerttemberg Cooperative State University Stuttgart}
```

As seen [here](https://rapla.dhbw-stuttgart.de/rapla?page=calendarlist) and [here](https://www.dhbw-stuttgart.de/zielgruppen/international-visitors/overview/)

JohnAZoidberg replied:
I just hope it doesn't look bad when it's that long. That's why I wanted to keep it short.

lschuermann commented:
Thanks for the fixes. Please excuse the extended delay, where I did not have a chance to get back to this. I think the requested change should be the last one.

lschuermann commented:
To be honest with you, I'm not quite sure why the CI build failed.

I can't find the Travis retry build option, and the API doesn't seem to work anymore. Maybe they removed it? So this could require a force push. :/

EDIT: Apparently, closing and opening should do the job.

lschuermann commented:
This is apparently a CI issue, and has nothing to do with your work. It builds reliably on two different systems, therefore I've temporarily removed the CI status requirement.

I've acknowledged the issue that the author field now overruns the title page. We'll fix this in another PR.

Thank you for your work and patience.
